okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -305,73 +313,70 @@ void removePendingStream(OkHttpClientStream pendingStream) {
   public void start(Listener listener) {
     this.listener = Preconditions.checkNotNull(listener, "listener");
 
-    frameWriter = new AsyncFrameWriter(this, serializingExecutor);
+    frameWriter = new AsyncFrameWriter(this, executor);
     outboundFlow = new OutboundFlowController(this, frameWriter);
 
-    // Connecting in the serializingExecutor, so that some stream operations like synStream
-    // will be executed after connected.
-    serializingExecutor.execute(new Runnable() {
+    // Connecting in another thread, do not block start.
+    executor.execute(new Runnable() {
       @Override
       public void run() {
         if (isForTest()) {
+          connectingCallback.run();
           clientFrameHandler = new ClientFrameHandler(testFrameReader);
           executor.execute(clientFrameHandler);
-          connectedCallback.run();
           frameWriter.becomeConnected(testFrameWriter, socket);
+          synchronized (lock) {
+            maxConcurrentStreams = Integer.MAX_VALUE;
+          }
+          startPendingStreams();
+          connectedFuture.set(null);
           return;
         }
+
         BufferedSource source;
         BufferedSink sink;
-        Socket sock;
```

Comment:
The socket field isn't used any more. Delete it and just use a local variable here.